staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -130,6 +132,9 @@ type GenericAPIServer struct {
 	swaggerConfig *swagger.Config
 	openAPIConfig *openapicommon.Config
 
+	// Enables updating OpenAPI spec using update method.
+	OpenAPIService *openapi.OpenAPIService
```

Comment:
That could work for OpenAPI spec, though I assume you can modify schema for a custom resource without adding/removing the APIService, right? In that case we may need a separate watch for OpenAPI spec (or modify discovery watch to support it)